osu.Game.Tournament/IPC/FileBasedIPC.cs

```diff
@@ -146,54 +155,135 @@ public Storage LocateStableStorage()
                 Logger.Error(e, "Stable installation could not be found; disabling file based IPC");
             }
 
-            return Storage;
+            return IPCStorage;
         }
 
+        public static bool CheckExists(string p) => File.Exists(Path.Combine(p, "ipc.txt"));
+
         private string findStablePath()
         {
-            static bool checkExists(string p) => File.Exists(Path.Combine(p, "ipc.txt"));
+            if (!string.IsNullOrEmpty(readStableConfig()))
+                return StableInfo.StablePath.Value;
```

Comment:
I find the fact that this returns a path only in this one case, and `null` in all others, really strange. This API is unintuitive to use.

If I got a `null` path from a method called `findStablePath()`, I'd assume that a stable path wasn't found - which is not the case, it means that `FileBasedIPC` found the path on its own, which is really counterintuitive IMO.

Reply:
> If I got a null path from a method called findStablePath(), I'd assume that a stable path wasn't found - which is not the case, it means that FileBasedIPC found the path on its own, which is really counterintuitive IMO.

I was very surprised when I first read this, because it was intended to also return the path for all other methods.  I then noticed that I committed a null I used to test a failure case (namely CI failing). This was very sloppy on my part and I will revert that right away.